### DISCUSSION TITLE
Encryption Choice

### DISCUSSION THREAD

**Discussion by tuckplace** [cs-utulsa/Encrypted-Chat-Service#2] [General]
Hey guys,

What do we all think for what type of encryption we will use? Any suggestions? For this project, we plan to use TCP to communicate and then we encrypt the payload. Before messages can be sent, we need a handshake to make the encryption dynamic. Then, we need to use this for encrypting our payload.

Please put any suggestions on this thread so we can discuss.

**Comment by doberzan** [comment]
My recommendation is implementing an asymmetric handshake/key exchange with the ED25519 algorithm, then use symmetric data encryption with AES-CBC/AES-EAX with HMAC-SHA256 as the message integrity verification

**Reply by tuckplace** [reply]
Sounds good to me, but I question what the load would be? Could a low power system handle this?

**Comment by tuckplace** [comment]
Going with ECC encryption
